### ISSUE TITLE
Some draft datasets in Harvard Dataverse repository database have version numbers?

### ISSUE THREAD

**Issue by jggautier** [IQSS/dataverse.harvard.edu#102]
In the Harvard Dataverse repository database, it looks like there are at least 251 datasets with draft versions that each have a major and minor version number. Is this an error in the database? Will those values be updated when those dataset owners publish their draft versions and choose how the version is updated (a major or minor version update)?

To see what I mean, run this query on the Harvard Dataverse repository database to see the 2423 draft versions (as of this writing) of the repository's published datasets:

```
select identifier, createtime, versionnumber, minorversionnumber, versionstate, dataverse.name
from datasetversion
join dvobject on dvobject.id = datasetversion.dataset_id
join dataverse on dataverse.id = dvobject.owner_id
where 
  publicationdate is not null and 
  versionstate = 'DRAFT'
order by createtime
```

For most of the versions, the values in the versionnumber and minorversionnumber columns are null. I've always thought this was because the user hasn't decided yet if the new version would be a new major version or a new minor version. Once the user publishes the dataset and decides how the version number should be updated, the database will record the new published version's version number in the versionnumber and minorversionnumber columns.

But for at least 251 of the draft versions in the results of that query (of the 2424 total), the versionnumber and minorversionnumber columns are not null. Run the same query as above but **where versionnumber is not null**:

```
select identifier, createtime, versionnumber, minorversionnumber, versionstate, dataverse.name
from datasetversion
join dvobject on dvobject.id = datasetversion.dataset_id
join dataverse on dataverse.id = dvobject.owner_id
where 
  publicationdate is not null and 
  versionstate = 'DRAFT' and
  versionnumber is not null
order by createtime
```

**Comment by djbrooke** [comment]
Thanks @jggautier - are all or most of these older? Is this possibly a migration artifact or is this something that's still happening with some regularity?

**Comment by jggautier** [comment]
Looks like 2 of the drafts were created earlier this month, 8 from 2020, 10 from 2019, one from 2018, and the rest were created between 2010 and 2015 (all of those have been assigned major versions).

Also not sure if this is relevant, but the ones from 2010-2015 have all been given major versions, and the rest are either major or minor versions.

**Comment by qqmyers** [comment]
The one example I see on a QDR test server is for a deaccessioned dataset - any possibility its related to deaccession?

**Comment by jggautier** [comment]
Hmmm, I'm not sure. Some of the datasets have deaccessioned versions but most don't, unless maybe for each dataset someone tried to deaccession a version and it failed for some reason, so there's no record of a deaccessioned version in the datasetversion table?

This query is saying that 37 of those 251 datasets have one or more deaccessioned versions so the other datasets don't have deaccessioned versions:

```
select distinct identifier
from dvobject
join datasetversion on datasetversion.dataset_id = dvobject.id
where 
   dvobject.id in(
      select dvobject.id
      from datasetversion
      join dvobject on dvobject.id = datasetversion.dataset_id
      join dataverse on dataverse.id = dvobject.owner_id
      where publicationdate is not null and versionstate = 'DRAFT'
      and versionnumber is not null
      )
   and versionstate = 'DEACCESSIONED'
```

**Comment by qqmyers** [comment]
It looks like the version is set in the PublishDatasetCommand and we now finish publication asynchronously in FinalizeDatasetPublicationCommand, which is where the file validation and globalID checks occur, so perhaps these are examples where the Publish command completed and the Finalize later failed.

**Comment by jggautier** [comment]
Thanks for helping look into this a bit! Would that mean that the depositors meant to publish these draft versions?

**Comment by qqmyers** [comment]
Possibly? There could still be other ways to get this (DDI imports?) and whether failures would have resulted in locks staying around would have depended on which release was running, but a failure that the user didn't follow-up on seems reasonable.

**Comment by jggautier** [comment]
Just re-ran the query for finding dataset draft versions that have version numbers and it returned 250. Not sure if that's 250 of the 251 returned in February.

In any case, if it's possible that people tried to publish these draft versions but they didn't publish, I think they should be included in a broader effort by the repository to figure out if depositors have left unpublished datasets and dataset versions unpublished intentionally.

Maybe something to consider @sbarbosadataverse as part of "site maintenance"?

**Comment by jggautier** [comment]
The dataset at https://dataverse.harvard.edu/dataset.xhtml?persistentId=doi:10.7910/DVN/YHXE2P is one of the 250 datasets with a draft version that's already been assigned a minor version number (1.1) in the repository's database:

<img width="1056" alt="Screen Shot 2021-06-02 at 11 32 27 AM" src="https://user-images.githubusercontent.com/18374574/120509092-3bc1cf00-c396-11eb-9c11-52ed2f6bedb4.png">

The depositor let us know that it's not actually a dataset (it used to be an example dataset for the collection's depositors to follow) and gave us the okay to remove it if needed. Before deaccessioning it, I was able to not only publish the draft version but publish it as a major version. I'm not sure how this change is represented in the database now - is the row in the datasetversion table overwritten? But this is evidence to me that this issue of draft versions being given version numbers won't prevent the depositors of these 250 datasets from publishing their draft versions or publishing them as major versions (when the draft version has been assigned a minor version number).

**Comment by jggautier** [comment]
I checked this issue again by re-running the query to find draft dataset versions with version numbers, and I found a couple of new ones since I last updated this GitHub issue in June 2021, so it's still happening.

I know that the recent discussion in the Google Group about producing logs for Make Data Count isn't describing what's going on in this GitHub issue, but it makes me wonder if these draft versions with version numbers might have any adverse affect on the creation of logs for the Harvard repository. @landreev recently spoke about creating those logs for the Harvard repo, and it didn't sound like there were any errors when creating them. Is that right?

**Comment by jggautier** [comment]
I ran the database query to find draft dataset versions with version numbers and I found seven new ones since I last updated this GitHub issue in Nov 2022, including a draft dataset version that was created earlier this month, so it's still happening.

But I haven't heard that this has prevented depositors from editing and publishing new versions of their deposits, and I haven't heard that this has affected the Make Data Count logs being processed for Harvard Dataverse.

So I'm going to close this GitHub issue.
